src/include/common/random_engine.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <mutex>
+#include <random>
+
+#include "common/assert.h"
+
+namespace kuzu {
+
+namespace main {
+class ClientContext;
+}
+
+namespace common {
+
+struct RandomState;
+
+struct RandomEngine {
+    explicit RandomEngine(int64_t seed = -1);
```

Comment:
I see we only create RandomEngine using zero parameter. Why do we keep this optional argument?